### PR TITLE
Improve the docker image for nexus-configurator.

### PR DESCRIPTION
**Pull request by janrotter**
This pull request introduces two changes. The first one instructs the nexus_configurator script to use the groovy scripts from pkg_resources, making it independent from the current working directory. The second commit adjusts the Dockerfile to use a multi-stage build (https://docs.docker.com/develop/develop-images/multistage-build/), allowing the image to be built with the Dockerfile only, so the dockerhub automatic builds can be used instead of Travis CI.

Please see the commit messages for more details.

### CODE REVIEW

**Comment by mikebryant** [comment]
I would note that we aren't going to switch from Travis to dockerhub automated builds - that would remove the pre-merge testing we currently have. Doesn't mean we can't accept the changes though

**Comment by mikebryant** [comment]
I think this is ok, but @stuart-warren can you take a look too?

File: Dockerfile [hunk 1]
```diff
@@ -1,7 +1,15 @@
...
+    ./build.sh
 
+FROM python:3.6-slim
 COPY . /app
 WORKDIR /app
-RUN pip install pipenv && pipenv install
-ENV SHELL=/bin/sh
-ENTRYPOINT ["pipenv", "run", "nexus_configurator"]
+COPY --from=0 /tmp/nexus_configurator/groovy /app/nexus_configurator/groovy
+RUN pip install .
```
Comment (stuart-warren) [inline Dockerfile]: this command doesn't obey the [Pipfile.lock](https://github.com/ocadotechnology/nexus-configurator/blob/master/Pipfile.lock) version of dependencies

**Review by stuart-warren — CHANGES_REQUESTED** [review]
other than that, LGTM

**Comment by janrotter** [comment]
@mikebryant What kind of testing are you referring to? I don't see any unit tests in the repository.

@stuart-warren Could you please take a look at the latest commit?

File: build.sh [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 release="v2.1.2"
 rm -rf nexus3-*
+rm -rf nexus_configurator/groovy
 curl -L https://github.com/ansible-ThoTeam/nexus3-oss/archive/${release}.tar.gz | tar xz
 mv nexus3-* nexus3-oss
-ln -s nexus3-oss/files/groovy/ groovy
+ln -s ../nexus3-oss/files/groovy nexus_configurator/groovy
```
Comment (stuart-warren) [inline build.sh:8]: sorry, this won't work with the current travis build as travis currently calls this script and then builds the image

**Review by stuart-warren — CHANGES_REQUESTED** [review]

**Comment by mikebryant** [comment]
@janrotter The basic testing that the docker build will work. Travis allows us to verify that prior to merge.

**Review by stuart-warren — APPROVED** [review]

**Pull request merged**